src/Plugins/Telemetry/Collectors/BlockchainMetricsCollector.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// BlockchainMetricsCollector.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.IEventHandlers;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.Plugins.Telemetry.Metrics;
+using Neo.SmartContract.Native;
+using System.Diagnostics;
+
+namespace Neo.Plugins.Telemetry.Collectors
+{
+    /// <summary>
+    /// Collects blockchain-related metrics including block height, sync status, and block processing times.
+    /// </summary>
+    public sealed class BlockchainMetricsCollector : ICommittedHandler, IDisposable
+    {
+        private readonly NeoSystem _system;
+        private readonly string _nodeId;
+        private readonly string _network;
+        private readonly Stopwatch _blockPersistStopwatch = new();
+        private DateTime _lastBlockTime = DateTime.UtcNow;
+        private bool _disposed;
```

Comment:
```suggestion
        private volatile bool _disposed;
```

`_disposed` should be volatile if it has concurrent acesses.